src/Forms/Form.php

```diff
@@ -317,6 +320,51 @@ public function __construct(
         $this->setupDefaultClasses();
     }
 
+    /**
+     * Make the form require sudo mode, which will make the form readonly and add a sudo mode password field
+     * unless the current user previously activated sudo mode.
+     */
+    public function requireSudoMode(): void
+    {
+        // Check that the current request handler for the form is one that's used
+        // in an admin context where sudo mode makes sense
+        $classes = [
+            LeftAndMain::class,
+            GridFieldDetailForm_ItemRequest::class,
+        ];
+        $enableSudoMode = false;
+        foreach ($classes as $class) {
+            if (class_exists($class) && is_a($this->getController(), $class)) {
+                $enableSudoMode = true;
+                break;
+            }
+        }
+        if (!$enableSudoMode) {
+            return;
+        }
+        // Check if sudo mode is currently enabled
+        $service = Injector::inst()->get(SudoModeServiceInterface::class);
+        $session = $this->getRequest()->getSession();
+        if ($service->check($session)) {
+            return;
+        }
+        // If sudo mode is not active, make the form readonly and add a sudo mode password field
+        $this->makeReadonly();
+        $name = SudoModePasswordField::FIELD_NAME;
+        $fields = $this->Fields();
+        if ($fields->fieldByName($name)) {
+            $fields->removeByName($name);
+        }
+        $field = SudoModePasswordField::create($name);
+        $field->setForm($this);
+        $first = $fields->first();
+        if ($first) {
+            $fields->insertBefore($first->getName(), $field);
+        } else {
+            $fields->push($field);
+        }
```

Comment:
```suggestion
        $fields->unshift($field);
```
We shouldn't need to explicitly set form, I think? If that is needed here for some reason, we probably need an inline code comment explaining why because that's highly unusual.

Also, we can just `unshift` directly to the top of the fieldlist.

Reply:
Updated to use unshift()

setForm() is not unusual at all e.g. FieldList $fields->setForm($this) is called in Form::__construct() which will loop $fields and call $field->setForm($this);

I don't see why it needs a comment fields normally get attached for their parent form

Reply:
It's unusual to _manually_ call `setForm()` when adding a field to a fieldlist. I don't think I've ever seen that - you usually just add the field to the fieldlist and let the form/fieldlist/field relations sort themselves out

Reply:
This still needs to be addressed. If setting the form here is necessary for some reason, please add a code comment explaining why, as I believe that should be handled automatically by adding the field to a fieldlist which belongs to a  form.

Reply:
Have added a code comment